harvester/harvester/tasks/preview.py

```diff
@@ -0,0 +1,76 @@
+from PIL import Image
+import os
+
+from celery.exceptions import SoftTimeLimitExceeded
+from django.conf import settings
+from django.core.files.storage import default_storage
+
+
+from core.models import ChromeScreenshotResource, Document
+from harvester.celery import app
+from core.utils.resources import serialize_resource
+
+
+THUMBNAIL_SIZES = [(400, 300), (200, 150)]
+
+
+@app.task(
+    name="generate_browser_preview",
+    soft_time_limit=60,
+    autoretry_for=(SoftTimeLimitExceeded,),
+    default_retry_delay=10
+)
+def generate_browser_preview(document_id):
+    document = Document.objects.get(pk=document_id)
+    screenshot_location = f"{settings.BASE_DIR}/screenshot-{document_id}.png"
+    resource = ChromeScreenshotResource()
+    resource.run(document.properties["url"], screenshot=screenshot_location)
+
+    if resource.success:
+        bucket_folder_path = f"previews/{document.id}"
+        create_thumbnails(screenshot_location, document.id)
+        store_previews(bucket_folder_path, document.id)
+        update_document(document, resource, bucket_folder_path)
+        resource.close()
+        remove_files_from_filesystem(document.id)
+
+
+def store_previews(destination_location, document_id):
+    with open(f"{settings.BASE_DIR}/screenshot-{document_id}.png", "rb") as file:
+        default_storage.save(
```

Comment:
Niet omdat het blocking is, maar om het even te agenderen. Ik kan me voorstellen dat het een goed idee is om de bestanden wat directer met de Resource te koppelen. Als de resource dan invalidate, dan kunnen we makkelijk de previews verwijderen. Je kunt dan queries uitvoeren zoals: delete alle documenten, die in deze deze dataset zitten, waar de index niet promoted is. Dan ruim je in 1x ook alle files op S3 op. In plaats van dat je met AWS CLI moet itereren en dan database checks moet doen om te kijken of id's in bestandsnamen bestaan. Het kan zeker, maar misschien wat complexer dan directe koppeling met de Resource. Relevant voor die ticket over invalidatie, die we vanochtend besproken hebben. Dit wel mergen lijkt mij.